### ISSUE TITLE
Act jupyter as a simple server

### ISSUE THREAD

**Issue by IcarusSO** [jupyter/help#118]
Is a possible to act jupyter notebook as a simple server such that
www.mynotebook.com:8888/api/get_data.py
allowing jupyter to execute get_data.py?

I think this is significantly useful to integrated with other programs, such as google spreadsheet which have great user interface.
It can save the development time especially for prototyping and for small team which have no many senior software developers.

**Comment by takluyver** [comment]
At present, no, that is not something you can do. What sort of workflow are you envisaging?

**Comment by bollwyvl** [comment]
Right, at present to get some new route on 8888, you would have to create a server_extension, which would have to be loaded at start time.

Your example is a little scary, as the notebook server should _never_ be run on a publicly-accessible host/port unless behind an authentication/containerization mechanism, as then anyone can run any code they want! A compromise is to use [kernel-gateway's http mode](http://jupyter-kernel-gateway.readthedocs.io/en/latest/http-mode.html) which can wrap up a notebook (but not an arbitrary python file) in a REST service, and _can_ be safe to run publicly... if you trust the notebook!

For local development, inside a running kernel, you can start anything you want! So you could imagine having two notebooks (therefore two kernels) open.
- in the first notebook, start your server (flask is probably easier than tornado for the general case)
- in the second notebook, use your code that talks to the other server

with subprocesses or maybe threads, you could do all of this in one kernel, and even use ipywidgets to make nice little buttons to start and stop the server process.

**Comment by IcarusSO** [comment]
@bollwyvl Thanks for your reply. Kernel-gatway's http requires specifying the notebook, which I think may be a bit inconvenience  than creating a server_extension.
I agree ipywidgets are cool but I prefer to use google spreadsheet because it is easier to be handled by others.
Of course, it must be run behind an authentication.
I think creating server_extension is the best solution and easy be handled by other programmers using different languages. I may fork the repository and implement this feature. :)
